api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -218,4 +218,7 @@ message TcpProxy {
   // The interval must be at least 1ms.
   google.protobuf.Duration access_log_flush_interval = 15
       [(validate.rules).duration = {gte {nanos: 1000000}}];
+
+  // If set to true, access log will be flushed when the TCP proxy has established connection with the upstream.
```

Comment:
_established a connection_

Reply:
Also, it might be surprising that it's not logged when TCP proxy fails to establish an upstream connection, so it's worth mentioning that.

Reply:
done